### PR TITLE
specify labels and annotations in AWX CR template

### LINKED ISSUE #16: Add ability to specify labels for "default" AWX template
### Feature Summary

"migrated" from https://github.com/ansible/awx-operator/issues/1820

At the moment there is possibility to specify additional_labels for child objects of AWX object, which si neat, but if someone want to use this functionality with default helm awx template, there is no way to specify labels for the AWX object itself (https://github.com/ansible/awx-operator/blob/devel/.helm/starter/templates/awx-deploy.yaml).
Or at least I dont know how, so I've used kustomize to patch the object. This could be easily avoided.

If no labels are specified for AWX object, then additional_labels are not being applied.

Please add ability to specify labels, when using default AWX deploy template.

### PR DESCRIPTION
**Pull request by miles-w-3**
Closes #16

### CODE REVIEW

**Review by schen1 — COMMENTED** [review]
I would add the labels and annotations under the current documentation:
https://github.com/ansible-community/awx-operator-helm?tab=readme-ov-file#awx

Otherwise LGTM

**Comment by miles-w-3** [comment]
@schen1 done

**Comment by oraNod** [comment]
@miles-w-3 Hey, I know it's been sitting here for a while. Apologies for taking so long to get back to your PRs. What do you think about getting to a final verdict on the chart versioning so we can fix ci and then we merge this? Since this is a code change, not just docs, I'd like to get ci working before merge. It shouldn't take that long. Cheers.

**Comment by oraNod** [comment]
thanks for this one @miles-w-3

File: .helm/starter/templates/awx-deploy.yaml [hunk 1]
```diff
@@ -5,6 +5,14 @@ kind: AWX
 metadata:
   name: {{ .name }}
   namespace: {{ $.Release.Namespace }}
+  {{- with .labels }}
+  labels:
+    {{ . | toYaml }}
+  {{- end }}
+  {{- with .annotations }}
+  labels:
```
Comment (Jibux) [inline .helm/starter/templates/awx-deploy.yaml:13]: Typo: it should be `annotations`.

Comment (miles-w-3) [reply .helm/starter/templates/awx-deploy.yaml:13]: @Jibux you are correct, fixing in https://github.com/ansible-community/awx-operator-helm/pull/28

**Pull request merged**